package/nfs/rancher-nfs

```diff
@@ -73,116 +73,100 @@ create() {
         print_success
     fi
 }
 
-delete() {
-    if [ -z "${OPTS[created]}" ]; then
-        print_success
-        exit 0
-    fi
+attach() {
+    print_not_supported
+}
 
-    if [ -z "${OPTS[name]}" ]; then
-        print_error "name is required"
+mountdest() {
+    # default configuration
+    local host="$DEFAULT_HOST"
+    local exportDir="$DEFAULT_EXPORT"
+    local opts="$DEFAULT_MOUNT_OPTS"
+    local name="${OPTS[name]}"
+    local mountDir="$MNT_DEST"
+
+    # if host/export are set, switch to driver_opts
+    if [ ! -z "${OPTS[host]}" ] && [ ! -z "${OPTS[export]}" ]; then
+        host="${OPTS[host]}"
+        exportDir="${OPTS[export]}"
+        opts="${OPTS[mntOptions]}"
     fi
 
-    local mntDest="$(tmp_dir)"
-    mount_default_share "${mntDest}"
-
-    # delete sub-directory using name parameter to represent volume
-    local subDir="${mntDest}"/"${OPTS[name]}"
-    if [ -d "$subDir" ]; then
-        rm -rf "${subDir}"
+    # if pooling is enabled, alter export to point to subdirectory
+    if [ "${OPTS[pool]}" != "false" ]; then
+        exportDir="$exportDir/$name"
     fi
 
-    # now unmount
-    unmount_default_share "${mntDest}"
-
+    mount_nfs "$host" "$exportDir" "$mountDir" "$opts"
     print_success
 }
 
-attach() {
-    print_not_supported
-}
-
 detach() {
     print_not_supported
 }
 
-mountdest() {
-    local mntOptions=""
-    if [ ! -z "${OPTS[mntOptions]}" ]; then
-        mntOptions="-o ${OPTS[mntOptions]}"
-    fi
-
-    local mntSrc="${OPTS[host]}":"${OPTS[export]}"
-
-    # if user does not supply host:export, we will mount remote share host:export/name to mntDst
-    if [ -z "${OPTS[host]}" ] || [ -z "${OPTS[export]}" ]; then
-        # get host:export from environment variables
-        if [ -z "${NFS_SERVER}" ]; then
-            print_error "Failed: No environment variable NFS_SERVER found"
-        fi
-
-        if [ -z "${MOUNT_DIR}" ]; then
-            print_error "Failed: No environment variable MOUNT_DIR found"
-        fi
-
-        if [ -z "${OPTS[name]}" ]; then
-            print_error "Failed: name is required"
-        fi
-
-        if [ ! -z "${MOUNT_OPTS}" ]; then
-            mntOptions="-o ${MOUNT_OPTS}"
-        fi
-
-        mntSrc="${NFS_SERVER}":"${MOUNT_DIR}"/"${OPTS[name]}"
-    fi
-
-    # mount remote share if not mounted already
-    if [ $(ismounted "${MNT_DEST}") == 0 ] ; then
-        mkdir -p "${MNT_DEST}"
-        get_host_process_pid
-        local error
-        error=`nsenter -t ${TARGET_PID} -n mount ${mntOptions} ${mntSrc} ${MNT_DEST} 2>&1`
-        if [ $? -ne 0 ]; then
-            print_error "Failed mount ${mntOptions} ${mntSrc} ${MNT_DEST}: ${error}"
-        fi
-    fi
-
+unmount() {
+    unmount_nfs "$MNT_DEST"
     print_success
 }
 
-unmount() {
-    if [ $(ismounted "${MNT_DEST}") == 1 ] ; then
-        error=`umount "${MNT_DEST}" 2>&1`
-        if [ $? -ne 0 ]; then
-            print_error "Failed umount ${MNT_DEST}: ${error}"
-        fi
+delete() {
```

Comment:
I haven't tested this function because Rancher doesn't seem to execute the `delete` anymore (this is acceptable to me because it is far safer to make user manually delete data - just like PVs in a K8s StatefulSet). 

It would be nice if we could `docker volume rm <name>` stack/container scoped volumes, but apparently the volume reference is deleted so docker/rancher have no knowledge of the volume after stack deletion.

Fair warning!